plugins/org.eclipse.n4js/src/org/eclipse/n4js/parser/LazyTokenStream.java

```diff
@@ -187,15 +189,123 @@ public void fillBuffer(int lookAhead) {
 		}
 	}
 
+	/////////// START FEATURE 'hidden()'
+
+	@Override
+	protected int skipOffTokenChannels(int i) {
+		if (forbidHiddenTokens) {
+			return i;
+		} else {
+			return super.skipOffTokenChannels(i);
+		}
+	}
+
+	@Override
+	protected int skipOffTokenChannelsReverse(int i) {
+		if (forbidHiddenTokens) {
+			return i;
+		} else {
+			return super.skipOffTokenChannelsReverse(i);
+		}
+	}
+
+	boolean forbidHiddenTokens = false;
+
+	@Override
+	protected void doSetHiddenTokens(String... lexerRules) {
+		super.doSetHiddenTokens(lexerRules);
+		if (lexerRules == null || lexerRules.length == 0) {
+			Preconditions.checkArgument(lexerRules == null || lexerRules.length == 0,
+					"Setting the hidden channel dynamically is only allowed for empty list of arguments");
```

Comment:
Something seems amiss, here: given the condition of the if-statement, this `#checkArgument()` call never fails.